src/Backend/Modules/Blog/Engine/Model.php

```diff
@@ -488,58 +479,62 @@ public static function getCategoryId(string $title, string $language = null): in
         );
     }
 
-    /**
-     * Get all data for a given id
-     *
-     * @param int $id The Id of the comment to fetch?
-     *
-     * @return array
-     */
-    public static function getComment($id): array
+    public static function getComment(int $id): array
     {
-        return (array) BackendModel::getContainer()->get('database')->getRecord(
-            'SELECT i.*, UNIX_TIMESTAMP(i.created_on) AS created_on,
-             p.id AS post_id, p.title AS post_title, m.url AS post_url
-             FROM blog_comments AS i
-             INNER JOIN blog_posts AS p ON i.post_id = p.id AND i.language = p.language
-             INNER JOIN meta AS m ON p.meta_id = m.id
-             WHERE i.id = ? AND p.status = ?
-             LIMIT 1',
-            [(int) $id, 'active']
-        );
+        $repository = BackendModel::get('doctrine.orm.default_entity_manager')
+            ->getRepository(Comment::class);
+        $comment = $repository->find($id);
+
+        if ($comment instanceof Comment) {
+            $commentData = $comment->toArray();
+
+            // I know this is dirty, but as we don't have full entities yet we
+            // need to fetch the post separately and inject it into the comment
+            $postData = (array)BackendModel::getContainer()->get('database')
+                ->getRecord(
+                    'SELECT p.id AS post_id, p.title AS post_title, m.url AS post_url
+                     FROM blog_posts AS p
+                     INNER JOIN meta AS m ON p.meta_id = m.id
+                     WHERE p.id = ? AND p.status = ? AND p.language = ?
+                     LIMIT 1',
+                    [
+                        (int)$comment->getId(),
+                        'active',
+                        $comment->getLocale()->getLocale(),
+                    ]
+                );
+
+            return array_merge(
+                $commentData,
+                $postData
+            );
+        }
+
+        return [];
     }
 
-    /**
-     * Get multiple comments at once
-     *
-     * @param array $ids The id(s) of the comment(s).
-     *
-     * @return array
-     */
     public static function getComments(array $ids): array
     {
-        return (array) BackendModel::getContainer()->get('database')->getRecords(
-            'SELECT *
-             FROM blog_comments AS i
-             WHERE i.id IN (' . implode(', ', array_fill(0, count($ids), '?')) . ')',
-            $ids
+        $repository = BackendModel::get('doctrine.orm.default_entity_manager')
```

Comment:
idem

Reply:
fixed in cd9cef81ef540cf410bc31750966a00591aa06ae